### PR TITLE
Fateme

### PR DESCRIPTION
**Pull request by fatemeebrahimzadeh**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+VITE_API_READ_ACCESS_TOKEN = "eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiJmZmQ5MTBhNDQxNzI5NmQ5MDczZjJlNmUzZjNlMDk1NyIsInN1YiI6IjY0YTJjZDhiMTEzODZjMDBhZGM3OWNjNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.UCEqmnJ1SMmALiGWVThtFffX-_Kvj1qHkUlmjehNehk"
```
Comment (amRashidi) [inline .env:1]: add .env to gitignore
add .env.example instead
avoid sending your personal token to public repositories
VITE_API_READ_ACCESS_TOKEN = YOUR_PERSONAL_TOKEN

File: src/components/CategoryMovieItem.vue [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<script setup>
+defineProps(['src', 'name', 'rate'])
+</script>
+
+<template>
+  <div class="movie-item relative group">
+    <img class="flex-shrink-0 object-cover" :src="src" />
+    <div class="movie-item__hover">
+      <a class="button z-10" href="./single-movie.html"> READ MORE </a>
```
Comment (amRashidi) [inline src/components/CategoryMovieItem.vue:9]: href value

File: src/components/CategoryMovieItem.vue [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    </div>
+    <div class="movie-item__title-in">
+      <h6 class="movie-item__title-in__heading">
+        <a href="./single-movie.html">{{ name }}</a>
```
Comment (amRashidi) [inline src/components/CategoryMovieItem.vue:13]: same

File: src/components/FavoriteMoviesSearchBar.vue [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        class="block w-full bg-white border-[1px] border-black dark:border-0 dark:bg-dark-triority placeholder-text text-text rounded-sm px-5 py-3 placeholder:font-light mb-5 text-xs"
+        required
+        placeholder="Enter keywords"
+      />
+      <label class="text-sm" for="skills">Genres & Subgenres:</label>
+      <select
+        name="skills"
+        class="flex-shrink-0 inline-flex font-medium text-center w-full bg-white border-[1px] border-black dark:border-0 dark:bg-dark-triority dark:placeholder-text dark:text-text rounded-sm px-5 py-3 placeholder:font-light mb-5 text-xs"
+      >
+        <option class="form-option" selected value="">Enter to filter genres</option>
```
Comment (amRashidi) [inline src/components/FavoriteMoviesSearchBar.vue:19]: use v-for

File: src/components/FavoriteMoviesMain.vue [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  <main class="py-10">
+    <div class="container flex flex-col lg:flex-row gap-10">
+      <div class="lg:basis-2/3">
+        <filter-bar />
+        <!-- replace grid to show -->
+        <section id="movie-list" class="hidden movie-list gap-5 my-10">
+          <movie-list-card v-for="index in 5" :key="index" />
+        </section>
+        <section id="movie-grid" class="movie-grid my-10 grid gap-5 grid-cols-1 sm:grid-cols-2">
+          <movie-grid-item-v1 />
```
Comment (amRashidi) [inline src/components/FavoriteMoviesMain.vue:20]: use v-for

File: src/components/FavoriteMoviesSearchBar.vue [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        <option class="form-option" value="Action3">Action 3</option>
+        <option class="form-option" value="Action4">Action 4</option>
+        <option class="form-option" value="Action5">Action 5</option>
+      </select>
+      <label class="text-xs" for="range">Rating Range</label>
+      <select
+        name="skills"
+        class="flex-shrink-0 inline-flex font-medium text-center w-full bg-white border-[1px] border-black dark:border-0 dark:bg-dark-triority dark:placeholder-text dark:text-text rounded-sm px-5 py-3 placeholder:font-light mb-5 text-xs"
+      >
+        <option class="form-option" value="range">-- Select the rating range below --</option>
```
Comment (amRashidi) [inline src/components/FavoriteMoviesSearchBar.vue:31]: same

File: src/components/FavoriteMoviesSearchBar.vue [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        <option class="form-option" value="saab">-- Select the rating range below --</option>
+      </select>
+      <label class="text-xs" for="year">Release Year:</label>
+      <div class="flex flex-col sm:flex-row justify-between gap-5">
+        <select
+          name="skills"
+          class="flex-shrink-0 inline-flex font-medium text-center basis-[45%] bg-white border-[1px] border-black dark:border-0 dark:bg-dark-triority dark:placeholder-text dark:text-text rounded-sm px-5 py-3 placeholder:font-light mb-5 text-xs"
+        >
+          <option
+            class="bg-light-secondary dark:bg-dark-secondary text-black dark:text-text"
```
Comment (amRashidi) [inline src/components/FavoriteMoviesSearchBar.vue:43]: same

File: src/components/FavoriteMoviesSearchBar.vue [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+            value="number"
+          >
+            30
+          </option>
+        </select>
+        <select
+          name="skills"
+          class="flex-shrink-0 inline-flex font-medium text-center basis-[45%] bg-white border-[1px] border-black dark:border-0 dark:bg-dark-triority dark:placeholder-text dark:text-text rounded-sm px-5 py-3 placeholder:font-light mb-5 text-xs"
+        >
```
Comment (amRashidi) [inline src/components/FavoriteMoviesSearchBar.vue:70]: same

File: src/components/FavoriteMoviesSidebar.vue [hunk 9]
```diff
@@ -0,0 +1,32 @@
+<script setup>
+import ads1 from '@/assets/images/ads1.png'
+
+import FavoriteMoviesSearchBar from '@/components/FavoriteMoviesSearchBar.vue'
+</script>
+
+<template>
+  <aside class="flex flex-col gap-5 lg:basis-1/3">
+    <section class="search grid gap-5">
```
Comment (amRashidi) [inline src/components/FavoriteMoviesSidebar.vue:9]: use v-for

File: src/components/FilterBar.vue [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    <div
+      class="flex flex-col justify-center sm:flex-row sm:justify-between sm:items-center sm:p-1 font-light sm:border-r-[1px] sm:border-light-secondary sm:dark:border-dark-100 sm:basis-8/12 w-full h-full"
+    >
+      <span class="text-center">Found <span class="text-blue">1,608 movies</span> in total</span>
+      <span class="text-center">Sort by:</span>
+    </div>
+    <select
+      class="max-w-[150px] border-[1px] sm:basis-2/12 font-light flex-shrink-0 inline-flex flex-col sm:flex-row sm:items-center sm:h-full text-center sm:border-0 sm:border-r-[1px] border-light-secondary dark:border-dark-100 text-black dark:text-text bg-transparent"
+    >
+      <option class="bg-light-secondary dark:bg-dark-secondary" value="popularity">
```
Comment (amRashidi) [inline src/components/FilterBar.vue:14]: use v-for

File: src/components/FooterContent.vue [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+            <p class="font-light">
+              Call us:
+              <a class="text-white item-hover" href="#">(+01) 202 342 6789</a>
+            </p>
+          </div>
+        </section>
+        <section class="footer-section">
+          <h4 class="footer-section__header">Resources</h4>
+          <ul class="footer-section__ul">
+            <li>
```
Comment (amRashidi) [inline src/components/FooterContent.vue:29]: use v-for

File: src/components/HeaderSlider.vue [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+import { API_IMAGE_BASE_URL } from '@/constants/api-constants.js'
+
+import { Axios } from '@/utils/Axios.js'
+import { inject, onMounted, ref } from 'vue'
+let movies = ref([])
+
+const { updateLoadingState } = inject('loading-state')
+updateLoadingState(true)
+
+onMounted(async () => {
```
Comment (amRashidi) [inline src/components/HeaderSlider.vue:13]: move logic to a function

File: src/components/HeaderSliderMovieCard.vue [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+  return props.genre_ids.map((id) => {
+    const genre = genres.find((g) => g.id === id)
+    return genre && genre
+  })
+})
+
+function redirectToSingleMoviePage() {
+  router.push({
+    path: '/single-movie',
+    query: { ...props, computedGenres: computedGenres.value }
```
Comment (amRashidi) [inline src/components/HeaderSliderMovieCard.vue:29]: bad query naming

File: src/components/ProfileMain.vue [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+  <main class="py-10">
+    <div class="container flex justify-center items-center -mt-52 lg:-mt-48">
+      <section
+        class="signup-content bg-light-backgroundColor dark:bg-dark-secondary dark:border-2 border-dark-800 w-full flex flex-col items-center p-5"
+      >
+        <h1 class="hidden text-black text-2xl font-dosis">PROFILE</h1>
+
+        <ul class="flex flex-row justify-around w-full">
+          <li class="single-movie-tab hidden sm:block">
+            <a href="./favorite-movies.html"> FAVORITE MOVIES </a>
```
Comment (amRashidi) [inline src/components/ProfileMain.vue:11]: href!!

File: src/components/SignupPopup.vue [hunk 15]
```diff
@@ -0,0 +1,34 @@
+<script setup>
+import github from '@/assets/svgs/github.svg'
+</script>
+
+<template>
+  <!-- make display flex and remove hidden -->
```
Comment (amRashidi) [inline src/components/SignupPopup.vue:6]: remove comment

File: src/components/SingleMovieHeader.vue [hunk 16]
```diff
@@ -0,0 +1,23 @@
+<script setup>
+import HeaderNav from '@/components/HeaderNav.vue'
+import HeaderSearchBar from '@/components/HeaderSearchBar.vue'
+defineProps(['backdrop_src'])
+</script>
+
+<template>
+  <header
+    :style="{ backgroundImage: 'url(' + backdrop_src + ')' }"
```
Comment (amRashidi) [inline src/components/SingleMovieHeader.vue:9]: use backtick

File: src/components/SingleMovieMain.vue [hunk 17]
```diff
@@ -0,0 +1,264 @@
...
+  'rate',
+  'computedGenres',
+  'release_date',
+  'vote_count',
+  'overview',
+  'relatedMovies'
+])
+
+const year = computed(() => {
+  return props.release_date.split('-')[0]
```
Comment (amRashidi) [inline src/components/SingleMovieMain.vue:23]: what if release date be empty?

File: src/components/SingleMovieMain.vue [hunk 18]
```diff
@@ -0,0 +1,264 @@
...
+                <span class="text-white text-lg">{{ rate }}</span>
+                <span>/10</span>
+                <p class="text-black dark:text-blue">{{ vote_count }} Reviews</p>
+              </div>
+            </span>
+          </div>
+          <div class="sm:basis-3/4 flex items-center text-lg gap-2 p-2">
+            Rate This Movie:
+            <span>
+              <i class="fa-fw fa-xl fa-star text-yellow fa"></i>
```
Comment (amRashidi) [inline src/components/SingleMovieMain.vue:66]: v-for

File: src/router/index.js [hunk 19]
```diff
@@ -0,0 +1,18 @@
+import { createRouter, createWebHashHistory } from 'vue-router'
+import HomePage from '@/pages/HomePage.vue'
+import FavoritePage from '@/pages/FavoritePage.vue'
+import SingleMoviePage from '@/pages/SingleMoviePage.vue'
+import ProfilePage from '@/pages/ProfilePage.vue'
+
+const routes = [
```
Comment (amRashidi) [inline src/router/index.js:7]: use lazyloading: ()=> import(path)

**Review by amRashidi — CHANGES_REQUESTED** [review]
please recheck the whole code for hrefs, v-for usage

File: src/components/SignupPopup.vue (hunk 15 above)
Comment (fatemeebrahimzadeh) [reply src/components/SignupPopup.vue:6]: I need them for develop that element later

File: .env (hunk 1 above)
Comment (fatemeebrahimzadeh) [reply .env:1]: please give a resource about using .env.example , I dont have any idea how can i add it and searching isnt clearly

Comment (amRashidi) [reply .env]: we fix this issue on meeting

**Review by amRashidi — APPROVED** [review]

**Pull request merged**